apps/sms/test/marionette/lib/messages.js

```diff
@@ -64,11 +64,20 @@ var ParticipantsAccessor = require('./views/participants/accessors');
 
         Participants: new ParticipantsAccessor(client),
 
-        launch() {
+        launch(doNotWaitForAppToBecomeReady) {
           client.switchToFrame();
           client.apps.launch(ORIGIN_URL);
           client.apps.switchToApp(ORIGIN_URL);
-          client.helper.waitForElement(SELECTORS.appReady);
+
+          // In 99.99% cases we'd like to wait for the app to become fully ready
```

Comment:
Regarding your proposal ```Could we just assert that thread number is still increasing after entering another view(query the number before/after new message view and compare)?```

It seems possible, but again we can't use existing ```inboxView.conversations.length``` because under the hood it uses ```findElements``` + several ```getAttribute``` calls and knowing that brand new default Marionette search timeout is ```60000ms``` I'm pretty sure calling ```inboxView.conversations.length``` will wait for all threads to load :)

So we have two ways:

* Introduce new method on inbox view and inbox accessor (```getConversationCount``` or something) that will use tiny timeout for ```findElements``` - but not sure how robust it's and how it really works under the hood.

* Also introduce new methods as above, but with ```scriptWith``` that will use ```querySelectorAll(...thread).length``` under the hood - a bit hacky but may work.

So that is why I went simpler route with existing ```hasConversation``` that already has reduce timeout - but agree it maybe confusing and not the best solution :)

Do you have better idea or maybe I just missing something, @steveck-chung?